### ISSUE TITLE
Event listeners not removed

### ISSUE THREAD

**Issue by ad9242** [mi6/ic-ui-kit#791]
## Summary of the bug
Some components add event listeners, which are not correctly removed. The following list may not be exhaustive, so all components should be reviewed:

**Ic-side-navigation:** 
"transitionend" event is never removed

**ic-breadcrumb-group:**
collapsedBreadcrumbEl "click" event is never removed 

"transitionend" event is never removed

**ic-checkbox:**
"icChange" on associated textfield does not remove same bound function instance

**Ic-horizontal-scroll**
"focus", "mouseup", "mouseleave" events are never removed 

**ic-tab-context:**
"keydown" listeners for tabGroup not removing same bound function

## 🧐 Expected behaviour
Any event listeners added should also be removed correctly, to prevent memory leaks

## Additional Info
basically this is bad:
```
document.addEventListener("keydown", () => foo());
document.removeEventListener("keydown", () => foo());
```
as it does not remove the exact same function instance
 
instead you need to do the following:

```
keydownHandler() {
  foo();
}
document.addEventListener("keydown", this.keydownHandler);
document.removeEventListener("keydown", this.keydownHandler);
```
